qiskit/synthesis/evolution/qdrift.py

```diff
@@ -0,0 +1,110 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""QDrift Class"""
+
+from typing import List, Union, Optional, Callable, Tuple
+import numpy as np
+from qiskit.circuit.quantumcircuit import QuantumCircuit
+from qiskit.quantum_info.operators import SparsePauliOp, Pauli
+from qiskit.utils import algorithm_globals
+
+from .product_formula import ProductFormula
+
+# pylint: disable=invalid-name
+
+
+class QDrift(ProductFormula):
+    r"""The QDrift Trotterization method, which selects each each term in the
+    Trotterization randomly, with a probability proportional to its weight. Based on the work
+    of Earl Campbell in Ref. [1].
+
+    References:
+        [1]: E. Campbell, "A random compiler for fast Hamiltonian simulation" (2018).
+         `arXiv:quant-ph/1811.08017 <https://arxiv.org/abs/1811.08017>`_
+    """
+
+    def __init__(
+        self,
+        reps: int = 1,
+        insert_barriers: bool = False,
+        cx_structure: str = "chain",
+        atomic_evolution: Optional[
+            Callable[[Union[Pauli, SparsePauliOp], float], QuantumCircuit]
+        ] = None,
+    ) -> None:
+        r"""
+        Args:
+            reps: The number of times to repeat the Trotterization circuit.
+            insert_barriers: Whether to insert barriers between the atomic evolutions.
+            cx_structure: How to arrange the CX gates for the Pauli evolutions, can be
+                "chain", where next neighbor connections are used, or "fountain", where all
+                qubits are connected to one.
+            atomic_evolution: A function to construct the circuit for the evolution of single
+                Pauli string. Per default, a single Pauli evolution is decomopsed in a CX chain
+                and a single qubit Z rotation.
+        """
+        super().__init__(1, reps, insert_barriers, cx_structure, atomic_evolution)
+        self.sampled_ops = None
+
+    @property
+    def sampled_ops(self) -> List[Tuple[Pauli, float]]:
+        """returns the list of sampled Pauli ops and their coefficients"""
+        return self._sampled_ops
+
+    @sampled_ops.setter
+    def sampled_ops(self, sampled_ops: List[Tuple[Pauli, float]]) -> None:
+        """sets the list of sampled Pauli ops and their coefficients"""
+        self._sampled_ops = sampled_ops
```

Comment:
If neither the setter nor the getter do any logic, is there a reason to not have `self.sampled_ops` be a regular attribute?

Reply:
Otherwise there is no easy way to know which operations are used in the circuit. This algorithm is very similar to the Lie-Trotter product formula but instead of simulating all Pauli strings you randomly sample a group of them and simulate those.

Reply:
Sure, I just meant that `self._sampled_ops` isn't doing anything useful here - you could just rename `self._sampled_ops` to `self.sampled_ops` and remove these two properties, and it'd have identical behaviour.

Reply:
ah yes that's right, I've removed the property and left it as `self._sampled_ops`